### ISSUE TITLE
Lange Ladezeit bei Authentication ./Login

### ISSUE THREAD

**Issue by ThePholph** [selectline-software/selectline-api#173]
20.3.10

Hallo,

wir haben seid gestern das Problem bei einem unserer Kunden, dass die Anmeldung an der SL.API länger wie 30 sek. dauert und dadurch in einem Timeout endet. Es wurden keinerlei Veränderungen am System durchgeführt.

Danke.

Viele Grüße

Daniel

**Comment by MatthiasGuse** [comment]
Hallo,

hier ein paar Fragen dazu:

1. Steht diese Änderung in irgend einer Verbindung mit einem Update der SelectLine Installation?
2. Hat der Server auf dem die API Installiert ist Zugriff auf das Internet? Hat sich etwas an der Firewall geändert?
3. Ist auf dem Server eine Warenwirtschaft installiert? Wann ja, startet diese ebenfalls sehr lange?

Viele Grüße
Matthias

**Comment by ThePholph** [comment]
>     1. Steht diese Änderung in irgend einer Verbindung mit einem Update der SelectLine Installation?
           Es gab keine Änderungen.

>     2. Hat der Server auf dem die API Installiert ist Zugriff auf das Internet? Hat sich etwas an der Firewall geändert?
           Ja, es wurde für den Testzeitraum die Firewall inkl. Endpoint Protection deaktiviert.

>     3. Ist auf dem Server eine Warenwirtschaft installiert? Wann ja, startet diese ebenfalls sehr lange?
           Ja, läuft einwandfrei.

Hallo,

was ich noch vergessen habe: Sl.Mobile ist auf zwei Servern installiert, um die Shop-Schnittstelle vom Ticketsystem getrennt behandeln zu können. Was bisher aber keine Probleme machte (> 1 Jahr).

Vielen Dank.

Viele Grüße

Daniel

**Comment by MatthiasGuse** [comment]
Tritt denn das Problem auf beiden Servern auf?

Sind im Mobile Manager alle Tests "grün"?

Ist das Verhalten auch nachstellbar, wenn Sie sich per Swagger oder Postman an der API anmelden?

**Comment by ThePholph** [comment]
> Tritt denn das Problem auf beiden Servern auf?

    ja, aber nur wenn beide Sl.Mobile "Dienste" laufen und Verbindungen (Token) von den verschiedenen Anwendungen geladen wurden

> Sind im Mobile Manager alle Tests "grün"?

    ja

> Ist das Verhalten auch nachstellbar, wenn Sie sich per Swagger oder Postman an der API anmelden?

    ja, ich nutze dafür den REST-Client im Firefox oder CURL, was auf das gleiche rauskommt.

**Comment by VivienSchuboth** [comment]
Guten Tag,
dieses Timeout hängt höchstwahrscheinlich mit der Datenbank zusammen.
Daher würde ich Sie bitten zu prüfen,
- ob die Datenbank korrekt eingestellt ist und 
- ob der Mandant (der dem API-Benutzer zugeordnet wurde) noch existiert.

Ich würde Sie ebenfalls bitten, dieselbe Anmeldung (hinsichtlich Benutzer, Anmeldeart am SQL-Server, Mandant) über Ihre Warenwirtschaft auszuprobieren. 
Haben Sie dort eine Verzögerung oder sich öffnende Fenster?

Wenn durch diese Fragen keine Fehler erkenntlich sind, würde ich Sie bitten, die Logs zu prüfen bzw. uns diese zuzuschicken, um Sie Inhouse selbst zu prüfen.

Beste Grüße

**Comment by ThePholph** [comment]
Hallo,

wie oben bereits beschrieben tritt das Problem nur auf, wenn die SL-API auf zwei verschiedenen Server läuft (zwei verschiedene Anwendung nutzen diese, die unabhängig voneinander laufen sollen). Bisher ist das auch kein Problem gewesen (> 1 Jahr) es gab auch vor kurzem keine Updates bzgl. SL. 

Daher erübrigt sich Ihre Frage bzgl. korrekter Einstellungen. 

Vielen Dank.

Daniel Fuchs

**Comment by SusanneSchadenberg** [comment]
Hallo,
leider kann ich das gemeldete Verhalten intern nicht nachstellen, daher benötige ich von Ihnen noch ein paar Angaben.
Greifen Sie von den 2 verschiedenen Mobile Installationen auf eine SelectLine Installation zu oder auf 2 getrennte?
Wo liegt der DB Server?
Laufen Makros beim Start der Warenwirtschaft?
Gibt wichtige Einträge in den Logs?

Viele Grüße aus Magdeburg
Susanne

**Comment by ThePholph** [comment]
Hallo,

folgende Konstellation:

Server: X-Shop (Anwendung 1)
 - SL.Mobile Installation
 - WaWi-Client Installation

Server: X-RX01 (Anwendung 2)
 - SL.Mobile Installation
 - WaWi-Client Installation

Server: X-DB01 (nur ein Mandant und ein DB-Server)
 - SL.Server Installation inkl. SQL-Server

Es gibt keine Makros beim Starten und in den Logs ist leider nichts zu sehen.

Vielen Dank.

Daniel

**Comment by MatthiasGuse** [comment]
Hallo,

eine Installation von mehreren mobilen Anwendungen auf unterschiedlichen Servern gehört nicht zu unserem gewünschten Installationszenario. Dies hat unter anderem zur Folge, dass die Backendprozesse jedes angemeldeten Benutzers auf beiden Servern ausgeführt bzw. gestartet werden. Ob es noch weitere Seitenenffekte gibt, können wir aktuell nicht beurteilen.

Zur Version 22.1 wurde einige Verbesserungen beim Start und der Verbindungsstabilität des Backends vorgenommen. Ebenfalls ist der Timeout für ein Login einstellbar geworden. Siehe Doku zur Installation der mobilen Anwendungen.

Viele Grüße
